assignment2/assign2_CRUD.c

```diff
@@ -0,0 +1,193 @@
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+
+#define FILENAME "users.txt"
+#define MAX_LEN 100
+
+typedef struct {
+    int id;
+    char name[MAX_LEN];
+    int age;
+} User;
+
+void createFile();
+void addUser();
+void display();
+void update();
+void deleteDetail();
+int getChoice();
+
+int main(){
+    int choice;
+
+    createFile();
+
+    while (1)
+    {
+         printf("-- User Management System ---\n");
+        printf("1. Add User\n");
+        printf("2. Display Users\n");
+        printf("3. Update User\n");
+        printf("4. Delete User\n");
+        printf("5. Exit\n");
+        printf("Enter your choice: ");
+        choice=getChoice();
+
+        switch(choice){
+            case 1: addUser(); break;
+            case 2: display(); break;
+            case 3: update(); break;
+            case 4: deleteDetail(); break;
+            case 5: exit(0);
+            default: printf("Invalid choice please try again.\n");
+        }
+    }
+
+    return 0;
+    
+}
+
+void createFile(){
+    FILE *file=fopen(FILENAME,"a");
+    if(file==NULL){
+        perror("Error creating file");
+        exit(1);
+    }
+    fclose(file);
+}
+
+void addUser(){
+    FILE *file=fopen(FILENAME,"a");
+    if(file==NULL){
+        perror("Error opening file");
+        return;
+    }
+
+    User user;
+    printf("Enter ID: ");
```

Comment:
Add a check for duplicate user IDs before adding a new user. It helps us when we are parsing the data from file.